### ISSUE TITLE
Git repository might need some cleanup

### ISSUE THREAD

**Issue by dvehrs** [danielmiessler/SecLists#54]
I've got this cloned onto my system and the directory takes up 778M of which 380M is the .git directory.  This seems excessive.  Might it be worthwhile to run garbage collection and pruning to reduce this down a bit?  Or is there a valid reason to keep it all that I'm missing?

Maybe run something like what is suggested here:
https://stackoverflow.com/questions/2116778/reduce-git-repository-size

**Comment by danielmiessler** [comment]
Phenomenal point. We'll look into it.

I'll keep it open until we get it resolved.

**Comment by shel3over** [comment]
you can use 
`git clone --depth 1` 
to clone the current state only ,without all the history

**Comment by dvehrs** [comment]
That's a good trick, shel3over.

From a quick test.  An updated clone with full history was 764M but a fresh clone with --depth 1 was only 508M.  The difference appears to be in the .git folders (448M vs 191M).  

Maybe a note on the opening page to let users know this is a recommended way to clone?

**Comment by toxydose** [comment]
> you can use
> `git clone --depth 1`
> to clone the current state only ,without all the history

I suggest to add this note with the notification about over-sized .git directory  to README file before this issue will be not fixed.

**Comment by kingthorin** [comment]
Note: If you use `git clone --depth 1` it'll be less to transfer and have on disk, however, you also won't be able to push or open PRs (in my experience).

**Comment by vonProteus** [comment]
maybe [migrate](https://github.com/git-lfs/git-lfs/blob/master/docs/man/git-lfs-migrate.1.ronn) to [git lfs](https://git-lfs.github.com/)
that would speed up the cloning process

**Comment by molangning** [comment]
> maybe [migrate](https://github.com/git-lfs/git-lfs/blob/master/docs/man/git-lfs-migrate.1.ronn) to [git lfs](https://git-lfs.github.com/) that would speed up the cloning process

lfs has GiB bandwidth per month, and any additional bandwidth needs to be purchased.

**Comment by ItsIgnacioPortal** [comment]
> lfs has GiB bandwidth per month, and any additional bandwidth needs to be purchased.

To quote [the github docs](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage):

> When you download a file tracked with Git LFS, the total file size is counted against the repository owner's bandwidth limit.

and

> If you use more than 1 GiB of bandwidth per month without purchasing a data pack, Git LFS support is disabled on your account until the next month.

Also, if seclists were to use git LFS, everyone who wanted to clone the repo would also need to install git LFS on their system before cloning. See: https://docs.github.com/en/repositories/working-with-files/managing-large-files/collaboration-with-git-large-file-storage

Therefor, git LFS is not viable to be used on Seclists.

**Comment by drwetter** [comment]
> Therefor, git LFS is not viable to be used on Seclists

Lame excuse :-)  Github is just not made to serve large files -- that's what git LFS is for -- note the acronym stands for large files ;-) .  Why don't you give it a shot? I found it a good alternative, at least for another repo I used in the past.

Guess this should also help when cloning the repo with --depth=1 this should reduce the pack file size . It's a pain in the butt. Besides, not everyone needs the large files served in this repo. I would just put them to LFS and the ones who wants them just have to use LFS.

**Comment by ItsIgnacioPortal** [comment]
At this point in time, it's only possible to reduce project file size for clones made without the commit history (a clone made with `git clone --depth 1 https://github.com/danielmiessler/SecLists.git`). This could be achieved by deleting wordlists related to legacy services, merging duplicate wordlists, and deleting potentially useless wordlists.

The file size of a complete clone could only be reduced by either rewriting history, or completely deleting the commit history. Both of those are destructive operations which would certainly break existing git clones of Seclists, forcing complete re-cloning of every single instance of the repo. That makes this option completely unviable.

**Comment by kingthorin** [comment]
Seems reasonable to me, and I appreciate that you brought a solution/suggestion.

**Comment by drwetter** [comment]
I do not see the user part -- installing git LFS as an obstacle. For me, who never uses the large files here it's an obstacle to use this repo. And it got worse during the years. It's very useful for other stuff though. 

Also looking at it from the engineering side: git is not really suitable or at least nice for either binaries or large files. Those things clog a repo -- been through that.

For the server part: Yes, git LFS costs money unfortunately. There might be several solutions to this -- to be completed and amended:

* ask for funding: Github itself, openssf, ...
* ask for sponsors through github, other donations from users
* Not use git LFS but put it some place else. Overall it's by other means not much data. That could be somewhere a server which is mirrored to a couple of places.

**Comment by ItsIgnacioPortal** [comment]
> I do not see the user part -- installing git LFS as an obstacle. For me, who never uses the large files here it's an obstacle to use this repo. And it got worse during the years. It's very useful for other stuff though.
> 
> Also looking at it from the engineering side: git is not really suitable or at least nice for either binaries or large files. Those things clog a repo -- been through that.
> 
> For the server part: Yes, git LFS costs money unfortunately. There might be several solutions to this -- to be completed and amended:
> 
>     * ask for funding: Github itself, openssf, ...
> 
>     * ask for sponsors through github, other donations from users
> 
>     * Not use git LFS but put it some place else. Overall it's by other means not much data. That could be somewhere a server which is mirrored to a couple of places.

I don't think it would be a good idea to make the project financially dependant on user donations, because that would mean that the project would become impossible to clone if the funding ever ran out.

Additionally, git LFS bandwith is not cheap. Github offers git LFS bandwith in "data packs", and each "data pack" gives 50GB for a month at the cost of $5. Considering that a `--depth 1` clone of Seclists weighs 1.8GB, every single clone of seclists would cost $0.18 a month.

See: https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-git-large-file-storage/upgrading-git-large-file-storage

**Comment by ItsIgnacioPortal** [comment]
Significant cleanup has been done in recent commits. I believe this issue can finally be closed.
